### ISSUE TITLE
[vLLM plugin][Bug] CPU samling failure

### ISSUE THREAD

**Issue by mmanzoorTT** [tenstorrent/tt-xla#4331]
vLLM plugin create sampling metadata on `cpu` when `cpu_sampling` is enabled. However, model runner is always using same `sample_from_logits` function. This cause failure as the sampling data is created on cpu whereas logits are still on device. 
`sample_from_logits_cpu` moves data to cpu but keep the output on cpu which may cause failure later.
